### PR TITLE
define the "inner_mask" for locstream case

### PR DESCRIPTION
**Pull request by vindelico**
Define the `inner_mask` in `subset_shape()` for the locstream case.
`subset_shape()` would return all lat/lon inside `inner_mask` which is equivalent to the bounding box of the polygon, not the area inside the polygon.

### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes a bug in issue #288 ; `subset_shape(...)` would return all points in the bounding box without filtering the points inside the polygon.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] I have added my relevant user information to `AUTHORS.md`

* **What kind of change does this PR introduce?:** 
The inner_mask in `subset_shape()` is defined for the locstream case.

* **Does this PR introduce a breaking change?:** 
No.

* **Other information:**

### CODE REVIEW

**Review by aulemahal — CHANGES_REQUESTED** [review]
Could you add a test for this? I think `test_locstream` (https://github.com/roocs/clisops/blob/c49fcc3b86c304fe9dd3699ed3ad5abe943892d1/tests/core/test_subset.py#L877) could be modified so that the test polygon is not a square.

**Comment by Zeitsperre** [comment]
@vindelico Do you have any free time to work on/finalize this PR? I have use cases that could benefit from your changes!

**Review by aulemahal — APPROVED** [review]
The changed test seems to answer my initial suggestion. Looks good!

**Comment by vindelico** [comment]
> @vindelico Do you have any free time to work on/finalize this PR? I have use cases that could benefit from your changes!

Can you check HISTORY.rst and AUTHORS.rst, plz? - otherwise I don't have additional changes.

File: HISTORY.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ Bug Fixes
 ^^^^^^^^^
 * The ``tests`` folder has been flattened and namespace files haves been removed in order to prevent `pip` from recognizing the folder as its own package. (#296).
 * The contribution guidelines were duplicated in two locations and contained conflicting information. The guidelines have now been consolidated into a single location and updated to reflect package changes. (#296).
+* The `subset_shape()` in the `locstream case` (#288) returned all lat/lon inside `inner_mask` which is equivalent to the bounding box of the polygon, not the area inside the polygon. Fixed by defining the `inner_mask` in `subset_shape()` for the locstream case. (#292).
```
Comment (Zeitsperre) [inline HISTORY.rst:17]: Could you start a new section above v0.11.0?
```
v0.12.0 (unreleased)
--------------------

Bug Fixes
^^^^^^^^^
```

Comment (vindelico) [reply HISTORY.rst]: Done.

**Review by Zeitsperre — APPROVED** [review]

**Pull request merged**